modules/dns/dns_api.rb

```diff
@@ -108,5 +113,9 @@ def validate_reverse_dns_name!(name)
       validate_dns_name!(name, 'PTR')
       raise Proxy::Dns::Error.new("Invalid reverse DNS #{name}") unless name =~ /\.(in-addr|ip6)\.arpa$/
     end
+
+    def validate_sshfp_data!(data)
+      raise Proxy::Dns::Error.new("Invalid SSHFP data #{data}") unless data =~/^[0-9]\s+[0-9]\s+[0-9a-f]+$/i
+    end
```

Comment:
Nitpick, exclamation mark is usually used when method changes state or data. But it's incorrect also above, so let's keep it consistent.

Reply:
I'd say the method with ! simply "can cause surprise", some rails method use it to indicate they raise exception, some ruby methods changes internal state (hence the surprise) but there's no good rule of thumb.